tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -3028,6 +3030,15 @@ namespace TensileLite
                                           Hardware const& hardware,
                                           size_t          tiles) const
     {
+        // If K==0, run kernel as DP with Alpha=0 to skip main loop and apply beta*c
+        size_t z = 1;
+        for(size_t i = 0; i < problem.boundIndices().size(); ++i)
+        {
+            z *= problem.boundSize(i);
+        }
```

Comment:
```suggestion
    size_t z = std::reduce(std::begin(x), std::end(x), size_t(1), std::multiplies{});
```
This would require that we include the `numeric` header.

Reply:
Thanks for the note! This patch needs to go with release changes, but I'll look at possible code cleanup in the next patch. This particular loop happens in a bunch of places in the code.